client/src/containers/CourseContainer.tsx

```diff
@@ -22,7 +22,7 @@ function mapDispatchToProps(dispatch: any, ownProps: any): any {
     dispatch(loadCourse(ownProps.params.id));
     return {
         onClickLesson(id: number) {
-            dispatch(push('/lesson/' + id));
+            dispatch(push('/study/' + id));
```

Comment:
This should dispatch to whatever the edit link is, since the lesson will be newly created.